src/main/java/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -1,27 +1,37 @@
 package cinema.security;
 
 import cinema.model.User;
+import cinema.service.RoleService;
 import cinema.service.ShoppingCartService;
 import cinema.service.UserService;
+import java.util.Set;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     private UserService userService;
     private ShoppingCartService shoppingCartService;
+    private PasswordEncoder passwordEncoder;
+    private RoleService roleService;
 
     public AuthenticationServiceImpl(UserService userService,
-                                     ShoppingCartService shoppingCartService) {
+                                     ShoppingCartService shoppingCartService,
+                                     PasswordEncoder passwordEncoder,
+                                     RoleService roleService) {
         this.userService = userService;
         this.shoppingCartService = shoppingCartService;
+        this.passwordEncoder = passwordEncoder;
+        this.roleService = roleService;
     }
 
     @Override
     public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        user = userService.add(user);
+        user.setRoles(Set.of(roleService.getRoleByName("USER")));
+        user = userService.add(user, passwordEncoder);
```

Comment:
let's autowire passwordEncoder in UserServiceImpl class and not pass it from here

Reply:
done